src/platform/nix_impl/profiler.rs

```diff
@@ -0,0 +1,295 @@
+use std::os::raw::c_int;
+use std::time::SystemTime;
+
+use crate::backtrace::{Frame, Trace, TraceImpl};
+use smallvec::SmallVec;
+
+use nix::sys::signal;
+
+use crate::error::Result;
+use crate::profiler::PROFILER;
+use crate::{MAX_DEPTH, MAX_THREAD_NAME};
+
+pub fn register() -> Result<()> {
```

Comment:
How about impl register in a trait? See `/src/backtrace/mod.rs: trait Trace`. We'll need to tell the developers which functions are needed to add a new `Profiler` (for different platform, or implementation)